src/createAction.js

```diff
@@ -10,35 +9,30 @@ export default function createAction(type, payloadCreator = identity, metaCreato
     'Expected payloadCreator to be a function, undefined or null'
   );
 
-  const finalPayloadCreator = isNull(payloadCreator)
+  const finalPayloadCreator = payloadCreator === null || payloadCreator === identity
```

Comment:
1. re: overhead, this is called just once during when defining the action creator, so do we need to worry about it? If not, I would advise removing the `isFunction` call above and below as well for consistency, but would prefer to just keep as is.